diff --git a/VSSurvivalMod/Systems/WorldGen/Standard/ChunkGen/4.GenBlockLayers/GenDevastationLayer.cs b/VSSurvivalMod/Systems/WorldGen/Standard/ChunkGen/4.GenBlockLayers/GenDevastationLayer.cs
index eb5ff1a..4163ce5 100644
--- a/VSSurvivalMod/Systems/WorldGen/Standard/ChunkGen/4.GenBlockLayers/GenDevastationLayer.cs
+++ b/VSSurvivalMod/Systems/WorldGen/Standard/ChunkGen/4.GenBlockLayers/GenDevastationLayer.cs
@@ -52,11 +52,11 @@ namespace Vintagestory.ServerMods
             api.Event.PreventTerrainHeightSmoothing(SuppressSmoothing);
             api.Event.PlayerJoin += Event_PlayerJoin;
 
             if (TerraGenConfig.DoDecorationPass)
             {
-                api.Event.ChunkColumnGeneration(OnChunkColumnGeneration, EnumWorldGenPass.Terrain, "standard");
+                api.Event.ChunkColumnGeneration(OnChunkColumnGeneration, EnumWorldGenPass.Terrain2, "standard"); // Stratum: we use our generation sub-stage
                 api.Event.GetWorldgenBlockAccessor(OnWorldGenBlockAccessor);
             }
 
             distDistort = new SimplexNoise(new double[] { 14, 9, 6, 3 }, new double[] { 1 / 100.0, 1 / 50.0, 1 / 25.0, 1 / 12.5 }, api.World.SeaLevel + 20980);
         }
