diff --git a/VintagestoryLib/Vintagestory.Server/CmdDebug.cs b/VintagestoryLib/Vintagestory.Server/CmdDebug.cs
index b005ed7..4c20140 100644
--- a/VintagestoryLib/Vintagestory.Server/CmdDebug.cs
+++ b/VintagestoryLib/Vintagestory.Server/CmdDebug.cs
@@ -326,15 +326,15 @@ public class CmdDebug
 			}
 			if (runtimeSpawnConditions.MaxQuantity > 0 || (runtimeSpawnConditions.MaxQuantityByGroup != null && runtimeSpawnConditions.MaxQuantityByGroup.MaxQuantity > 0))
 			{
 				if (entityType.Color == null || entityType.Color == "")
 				{
-					((SKColor)(ref color))._002Ector((uint)(((int)random.NextInt64() & 0xFFFFFF) | 0x60000000));
+					color = new SKColor((uint)(((int)random.NextInt64() & 0xFFFFFF) | 0x60000000));
 				}
 				else
 				{
-					((SKColor)(ref color))._002Ector((uint)((ColorUtil.Hex2Int(entityType.Color) & 0xFFFFFF) | 0x60000000));
+					color = new SKColor((uint)((ColorUtil.Hex2Int(entityType.Color) & 0xFFFFFF) | 0x60000000));
 				}
 				list.Add(new Eproprs
 				{
 					Props = entityType,
 					Color = color
@@ -387,12 +387,12 @@ public class CmdDebug
 						flag4 = num4 >= runtime.MinY - 1f && num4 <= runtime.MaxY - 1f;
 						break;
 					}
 					if (flag3 && flag4)
 					{
-						int num5 = ColorUtil.ColorOverlay(bitmapExternal.bmp.GetPixel(4 + i, 4 + j).ToInt(), eproprs.Color.ToInt(), (float)(int)((SKColor)(ref eproprs.Color)).Alpha / 255f);
-						((SKColor)(ref val))._002Ector((uint)num5);
+						int num5 = ColorUtil.ColorOverlay(bitmapExternal.bmp.GetPixel(4 + i, 4 + j).ToInt(), eproprs.Color.ToInt(), (float)(int)eproprs.Color.Alpha / 255f);
+						val = new SKColor((uint)num5);
 						bitmapExternal.bmp.SetPixel(4 + i, 4 + j, val);
 					}
 				}
 			}
 		}
@@ -586,11 +586,11 @@ public class CmdDebug
 		{
 			server.doNetBenchmark = false;
 			StringBuilder stringBuilder = new StringBuilder();
 			foreach (KeyValuePair<int, int> packetBenchmarkByte in server.packetBenchmarkBytes)
 			{
-				SystemNetworkProcess.ServerPacketNames.TryGetValue(packetBenchmarkByte.Key, out var value);
+				string value = "packet#" + packetBenchmarkByte.Key;
 				int num = server.packetBenchmark[packetBenchmarkByte.Key];
 				stringBuilder.AppendLine(num + "x " + value + ": " + ((packetBenchmarkByte.Value > 9999) ? (((float)packetBenchmarkByte.Value / 1024f).ToString("#.#") + "kb") : (packetBenchmarkByte.Value + "b")));
 			}
 			stringBuilder.AppendLine("-----");
 			foreach (KeyValuePair<string, int> packetBenchmarkBlockEntitiesByte in server.packetBenchmarkBlockEntitiesBytes)
