diff --git a/VintagestoryLib/Vintagestory.Server/ServerSystemCompressChunks.cs b/VintagestoryLib/Vintagestory.Server/ServerSystemCompressChunks.cs
index a8590af..dd18d42 100644
--- a/VintagestoryLib/Vintagestory.Server/ServerSystemCompressChunks.cs
+++ b/VintagestoryLib/Vintagestory.Server/ServerSystemCompressChunks.cs
@@ -22,6 +22,8 @@ internal class ServerSystemCompressChunks(ServerMain server) : ServerSystem(serv
 
 	private List<int> clientIds = new List<int>();
 
+	private readonly List<BlockPos> stratumPlayerChunkPositions = new List<BlockPos>();
+
 	public override void OnPlayerJoin(ServerPlayer player)
 	{
 		lock (clientIdsLock)
@@ -59,7 +61,8 @@ internal class ServerSystemCompressChunks(ServerMain server) : ServerSystem(serv
 
 	private void FindFreeableMemory()
 	{
-		List<BlockPos> list = new List<BlockPos>();
+		List<BlockPos> list = stratumPlayerChunkPositions;
+		list.Clear();
 		lock (clientIdsLock)
 		{
 			foreach (int clientId in clientIds)
