diff --git a/VintagestoryLib/Vintagestory.Server/ServerSystemRelight.cs b/VintagestoryLib/Vintagestory.Server/ServerSystemRelight.cs
index 82aaf34..6529b4f 100644
--- a/VintagestoryLib/Vintagestory.Server/ServerSystemRelight.cs
+++ b/VintagestoryLib/Vintagestory.Server/ServerSystemRelight.cs
@@ -10,6 +10,8 @@ public class ServerSystemRelight : ServerSystem
 {
 	public ChunkIlluminator chunkIlluminator;
 
+	private readonly HashSet<long> stratumAffectedChunks = new HashSet<long>();
+
 	public ServerSystemRelight(ServerMain server)
 		: base(server)
 	{
@@ -63,7 +65,8 @@ public class ServerSystemRelight : ServerSystem
 		int x = task.pos.X;
 		int internalY = task.pos.InternalY;
 		int z = task.pos.Z;
-		HashSet<long> hashSet = new HashSet<long>();
+		HashSet<long> hashSet = stratumAffectedChunks;
+		hashSet.Clear();
 		if (task.absorbUpdate)
 		{
 			num = task.oldAbsorb;
