diff --git a/VintagestoryLib/Vintagestory.Server/ServerSystemCompressChunks.cs b/VintagestoryLib/Vintagestory.Server/ServerSystemCompressChunks.cs
index a8590af..dd18d42 100644
--- a/VintagestoryLib/Vintagestory.Server/ServerSystemCompressChunks.cs
+++ b/VintagestoryLib/Vintagestory.Server/ServerSystemCompressChunks.cs
@@ -20,10 +20,12 @@ internal class ServerSystemCompressChunks(ServerMain server) : ServerSystem(serv
 
 	private object clientIdsLock = new object();
 
 	private List<int> clientIds = new List<int>();
 
+	private readonly List<BlockPos> stratumPlayerChunkPositions = new List<BlockPos>();
+
 	public override void OnPlayerJoin(ServerPlayer player)
 	{
 		lock (clientIdsLock)
 		{
 			clientIds.Add(player.ClientId);
@@ -57,11 +59,12 @@ internal class ServerSystemCompressChunks(ServerMain server) : ServerSystem(serv
 		}
 	}
 
 	private void FindFreeableMemory()
 	{
-		List<BlockPos> list = new List<BlockPos>();
+		List<BlockPos> list = stratumPlayerChunkPositions;
+		list.Clear();
 		lock (clientIdsLock)
 		{
 			foreach (int clientId in clientIds)
 			{
 				if (server.Clients.TryGetValue(clientId, out var value) && value.State == EnumClientState.Playing)
