diff --git a/VintagestoryLib/Vintagestory.Server/ServerSettings.cs b/VintagestoryLib/Vintagestory.Server/ServerSettings.cs
index f0ccce2..e835a2c 100644
--- a/VintagestoryLib/Vintagestory.Server/ServerSettings.cs
+++ b/VintagestoryLib/Vintagestory.Server/ServerSettings.cs
@@ -1,10 +1,10 @@
 using Newtonsoft.Json;
 
 namespace Vintagestory.Server;
 
-[JsonObject(/*Could not decode attribute arguments.*/)]
+[JsonObject(MemberSerialization.OptIn)]
 public class ServerSettings
 {
 	public static ServerSettings instance = new ServerSettings();
 
 	[JsonProperty]
