diff --git a/VintagestoryLib/Vintagestory.Server/WorldGenHandler.cs b/VintagestoryLib/Vintagestory.Server/WorldGenHandler.cs
index 20f5814..f0ad9e1 100644
--- a/VintagestoryLib/Vintagestory.Server/WorldGenHandler.cs
+++ b/VintagestoryLib/Vintagestory.Server/WorldGenHandler.cs
@@ -13,30 +13,33 @@ public class WorldGenHandler : IWorldGenHandler
 
 	public List<MapRegionGeneratorDelegate> OnMapRegionNeighborsLoaded = new List<MapRegionGeneratorDelegate>();
 
 	public List<MapChunkGeneratorDelegate> OnMapChunkGen = new List<MapChunkGeneratorDelegate>();
 
-	public List<ChunkColumnGenerationDelegate>[] OnChunkColumnGen = new List<ChunkColumnGenerationDelegate>[6];
+	public List<ChunkColumnGenerationDelegate>[] OnChunkColumnGen = new List<ChunkColumnGenerationDelegate>[(int)EnumWorldGenPass.Done]; // Stratum:  Getting rid of the magic number
 
-	public Dictionary<string, WorldGenHookDelegate> SpecialHooks = new Dictionary<string, WorldGenHookDelegate>();
+    public Dictionary<string, WorldGenHookDelegate> SpecialHooks = new Dictionary<string, WorldGenHookDelegate>();
 
 	List<MapRegionGeneratorDelegate> IWorldGenHandler.OnMapRegionGen => OnMapRegionGen;
 
 	List<MapChunkGeneratorDelegate> IWorldGenHandler.OnMapChunkGen => OnMapChunkGen;
 
 	List<ChunkColumnGenerationDelegate>[] IWorldGenHandler.OnChunkColumnGen => OnChunkColumnGen;
 
 	public WorldGenHandler()
 	{
-		OnChunkColumnGen[1] = new List<ChunkColumnGenerationDelegate>();
-		OnChunkColumnGen[2] = new List<ChunkColumnGenerationDelegate>();
-		OnChunkColumnGen[3] = new List<ChunkColumnGenerationDelegate>();
-		OnChunkColumnGen[4] = new List<ChunkColumnGenerationDelegate>();
-		OnChunkColumnGen[5] = new List<ChunkColumnGenerationDelegate>();
-	}
-
-	public void WipeAllHandlers()
+        // Stratum start: Initialize the generator lists for all work stages
+        // regardless of how many there may be
+        // (Terrain..PreDone), excluding None (0) and Done
+        for (int pass = (int)EnumWorldGenPass.None+1; pass < (int)EnumWorldGenPass.Done; pass++)
+        {
+            OnChunkColumnGen[pass] = new List<ChunkColumnGenerationDelegate>();
+        }
+        // Stratum end
+    }
+
+    public void WipeAllHandlers()
 	{
 		OnMapRegionGen.Clear();
 		OnMapChunkGen.Clear();
 		for (int i = 0; i < OnChunkColumnGen.Length; i++)
 		{
