diff --git a/VSEssentials/Entity/AI/AiTaskBase.cs b/VSEssentials/Entity/AI/AiTaskBase.cs
index 72f34f0..1ce43dd 100644
--- a/VSEssentials/Entity/AI/AiTaskBase.cs
+++ b/VSEssentials/Entity/AI/AiTaskBase.cs
@@ -211,11 +211,11 @@ namespace Vintagestory.API.Common
             SetDefaultValues();
             if (taskConfig.Exists)
             {
                 // Note that the generic type here is only used for proving the target is a class rather than a struct,
                 // and does not limit which fields will be populated
-                JsonUtil.Populate(taskConfig.Token, this);
+                JsonUtil.Populate<AiTaskBase>(taskConfig.Token, this); // Stratum: constrain to base class fields only
             }
 
             Id = taskConfig["id"].AsString();
 
             int initialmincooldown = taskConfig["initialMinCoolDown"].AsInt(MinCooldownMs);
