diff --git a/VSEssentials/Entity/AI/AiTaskBase.cs b/VSEssentials/Entity/AI/AiTaskBase.cs
index 72f34f0..6bdd7aa 100644
--- a/VSEssentials/Entity/AI/AiTaskBase.cs
+++ b/VSEssentials/Entity/AI/AiTaskBase.cs
@@ -138,13 +138,10 @@ namespace Vintagestory.API.Common
         [Obsolete("Use WhenNotInEmotionStates (plural, array) instead")] // May be set as private later. Supports json backwards comaptibility.
         [JsonProperty]
         public string? WhenNotInEmotionState { get => WhenNotInEmotionStates?[0]; set => WhenNotInEmotionStates = value?.Split("|"); }
 
         protected long cooldownUntilMs;
-        protected int? minGeneration;
-        protected int? maxGeneration;
-
         protected double cooldownUntilTotalHours;
 
         protected WaypointsTraverser pathTraverser;
         protected EntityBehaviorEmotionStates? bhEmo;
 
@@ -211,23 +208,19 @@ namespace Vintagestory.API.Common
             SetDefaultValues();
             if (taskConfig.Exists)
             {
                 // Note that the generic type here is only used for proving the target is a class rather than a struct,
                 // and does not limit which fields will be populated
-                JsonUtil.Populate(taskConfig.Token, this);
+                JsonUtil.Populate<AiTaskBase>(taskConfig.Token, this);
             }
 
             Id = taskConfig["id"].AsString();
 
             int initialmincooldown = taskConfig["initialMinCoolDown"].AsInt(MinCooldownMs);
             int initialmaxcooldown = taskConfig["initialMaxCoolDown"].AsInt(MaxCooldownMs);
-
             cooldownUntilMs = entity.World.ElapsedMilliseconds + initialmincooldown + entity.World.Rand.Next(initialmaxcooldown - initialmincooldown);
 
-            minGeneration = taskConfig["minGeneration"].Exists ? taskConfig["minGeneration"].AsInt() : null;
-            maxGeneration = taskConfig["maxGeneration"].Exists ? taskConfig["maxGeneration"].AsInt() : null;
-
             timeout = TimeSpan.FromSeconds(taskConfig["timeoutSec"].AsDouble(defaultTimeoutSec));
 
             JsonObject animationCfg = taskConfig["animation"];
             if (animationCfg.Exists)
             {
@@ -319,12 +312,10 @@ namespace Vintagestory.API.Common
         protected virtual bool PreconditionsSatisfied()
         {
             if (WhenSwimming != null && WhenSwimming != entity.Swimming) return false;
             if (WhenInEmotionStates != null && !IsInEmotionState(WhenInEmotionStates)) return false;
             if (WhenNotInEmotionStates != null && IsInEmotionState(WhenNotInEmotionStates)) return false;
-            if (minGeneration != null && entity.WatchedAttributes.GetInt("generation") < minGeneration) return false;
-            if (maxGeneration != null && entity.WatchedAttributes.GetInt("generation") > maxGeneration) return false;
             if (!IsValidTemperature()) return false;
             if (!IsValidLightLevel()) return false;
             if (!IsInValidDayTimeHours(true)) return false;
             return true;
         }
