diff --git a/VintagestoryLib/Vintagestory.Server.Network/ServerNetManager.cs b/VintagestoryLib/Vintagestory.Server.Network/ServerNetManager.cs
index f33469f..7dca2f0 100644
--- a/VintagestoryLib/Vintagestory.Server.Network/ServerNetManager.cs
+++ b/VintagestoryLib/Vintagestory.Server.Network/ServerNetManager.cs
@@ -2,11 +2,10 @@ using System;
 using System.Net;
 using System.Net.Sockets;
 using System.Threading;
 using System.Threading.Tasks;
 using Vintagestory.API.Common;
-
 namespace Vintagestory.Server.Network;
 
 public class ServerNetManager : IDisposable
 {
 	public Socket Socket;
