diff --git a/VintagestoryApi/Server/Worldgen/EnumWorldGenPass.cs b/VintagestoryApi/Server/Worldgen/EnumWorldGenPass.cs
index f3ac51d..7a092c2 100644
--- a/VintagestoryApi/Server/Worldgen/EnumWorldGenPass.cs
+++ b/VintagestoryApi/Server/Worldgen/EnumWorldGenPass.cs
@@ -13,19 +13,27 @@ namespace Vintagestory.API.Server
         /// <summary>
         /// Nothing generated yet
         /// </summary>
         None = 0,
 
+        // Stratum start: split the Terrain into two stages
+        // this way we get a uniform load across each stage across threads
+
         /// <summary>
         /// Does not require neighbour chunks to exist. Should generates 3d rock terrain mostly. Default generators by execute order:
         /// 0 = Basic 3D Terrain (granite+rock)
+        /// </summary>
+        Terrain = 1,
+
+        /// <summary>
+        /// Doesn't require adjacent fragments to work. Generates the remaining terrain stages. Generators are run in the following order by default:
         /// 0.1 = Rock Strata
         /// 0.2 = Rivers
         /// 0.3 = Cave generator
         /// 0.4 = Block layers (soil, gravel, sand, ice, tall grass, etc.)
         /// </summary>
-        Terrain = 1,
+        Terrain2 = 2,
 
         // Beyond here
         // - Chunkloader ensures that all neighbour chunks are same or higher pass
 
 
@@ -35,37 +43,39 @@ namespace Vintagestory.API.Server
         /// 0.2 = Deposits (Ores, Peat, Clay, etc.)
         /// 0.3 = Worldgen Structures
         /// 0.4 = Above sealevel Lakes
         /// 0.5 = Worldgen Structures Post Pass
         /// </summary>
-        TerrainFeatures = 2,
+        TerrainFeatures = 3,
 
         /// <summary>
         /// Requires neighbour chunks. Default generators by execute order:
         /// 0.2 = Story structures. Creates exclusion zones for the other vegetation passes
         /// 0.5 = Block Patches, Shrubs and Trees
         /// 0.9 = Rivulets (single block water sources)
         /// 0.95 = Sunlight flooding only inside current chunk
         /// </summary>
-        Vegetation = 3,
+        Vegetation = 4,
 
         /// <summary>
         /// Requires neighbour chunks. Does the lighting of the chunk.
         /// 0 = Snow layer
         /// 0.95 = Sunlight flooding into neighbouring chunks
         /// </summary>
-        NeighbourSunLightFlood = 4,
+        NeighbourSunLightFlood = 5,
 
         /// <summary>
         /// Requires neighbour chunks. Nothing left to generate, but neighbour chunks might still generate stuff into this chunk
         /// 0.1 = Generate creatures
         /// </summary>
-        PreDone = 5,
+        PreDone = 6,
 
         /// <summary>
         /// Chunk generation complete. This pass is not triggered as an event. 
         /// </summary>
-        Done = 6
+        Done = 7
+
+        // Stratum end
     }
 
 
 }
