diff --git a/VSEssentials/Systems/WorldGen/Standard/GenPartial.cs b/VSEssentials/Systems/WorldGen/Standard/GenPartial.cs
index 58aa710..14f2ccf 100644
--- a/VSEssentials/Systems/WorldGen/Standard/GenPartial.cs
+++ b/VSEssentials/Systems/WorldGen/Standard/GenPartial.cs
@@ -1,17 +1,20 @@
 using Vintagestory.API.MathTools;
 using Vintagestory.API.Server;
+using Vintagestory.ServerMods.NoObf;
 
 #nullable disable
 
 namespace Vintagestory.ServerMods
 {
     public abstract class GenPartial : ModStdWorldGen
     {
         protected ICoreServerAPI api;
         protected int worldheight;
         public int airBlockId = 0;
+        public int basaltBlockId; // Stratum: variable for basalt ID
+        public int lavaBlockId; // Stratum: variable for lav ID
 
         protected abstract int chunkRange { get; }
 
         protected LCGRandom chunkRand;
 
@@ -24,10 +27,13 @@ namespace Vintagestory.ServerMods
 
         public virtual void initWorldGen()
         {
             LoadGlobalConfig(api);
 
+            basaltBlockId = gcfg.basaltBlockId; // Stratum: finding the basalt block ID
+            lavaBlockId = gcfg.lavaBlockId;  // Stratum: finding the lsvs block ID
+
             worldheight = api.WorldManager.MapSizeY;
             chunkRand = new LCGRandom(api.WorldManager.Seed);
         }
 
 
