diff --git a/VintagestoryLib/Vintagestory.Client/GuiScreenSingleplayerModify.cs b/VintagestoryLib/Vintagestory.Client/GuiScreenSingleplayerModify.cs
index 582a9a2..21ea495 100644
--- a/VintagestoryLib/Vintagestory.Client/GuiScreenSingleplayerModify.cs
+++ b/VintagestoryLib/Vintagestory.Client/GuiScreenSingleplayerModify.cs
@@ -71,16 +71,16 @@ public class GuiScreenSingleplayerModify : GuiScreen
 			worldSeed = savegame.Seed;
 			playstylelangcode = savegame.PlayStyleLangCode;
 			ElementBounds elementBounds = ElementBounds.Fixed(0.0, 0.0, 330.0, 80.0);
 			ElementBounds elementBounds2 = elementBounds.BelowCopy().WithFixedHeight(35.0);
 			TextExtents textExtents = CairoFont.ButtonText().GetTextExtents(Lang.Get("Save"));
-			double num = ((TextExtents)(ref textExtents)).Width / (double)RuntimeEnv.GUIScale + 40.0;
+			double num = textExtents.Width / (double)RuntimeEnv.GUIScale + 40.0;
 			textExtents = CairoFont.ButtonText().GetTextExtents(Lang.Get("Customize"));
-			double num2 = ((TextExtents)(ref textExtents)).Width / (double)RuntimeEnv.GUIScale + 40.0;
+			double num2 = textExtents.Width / (double)RuntimeEnv.GUIScale + 40.0;
 			string text = Lang.Get("Create a new world with this world seed");
 			textExtents = CairoFont.WhiteSmallText().GetTextExtents(text);
-			double num3 = ((TextExtents)(ref textExtents)).Width / (double)RuntimeEnv.GUIScale + 40.0;
+			double num3 = textExtents.Width / (double)RuntimeEnv.GUIScale + 40.0;
 			string text2 = ((savegame.PlayStyleLangCode == null) ? Lang.Get("playstyle-" + savegame.PlayStyle) : Lang.Get("playstyle-" + savegame.PlayStyleLangCode));
 			GuiComposer composer = dialogBase("mainmenu-singleplayermodifyworld", -1.0, 550.0).AddStaticText(Lang.Get("Modify World"), CairoFont.WhiteSmallishText(), elementBounds).AddStaticText(Lang.Get("World name"), CairoFont.WhiteSmallishText(), elementBounds2).AddTextInput(elementBounds2.FlatCopy().WithAlignment(EnumDialogArea.RightFixed).WithFixedOffset(0.0, -3.0)
 				.WithFixedSize(470.0, 30.0), null, null, "worldname")
 				.AddStaticText(Lang.Get("Filename on disk"), CairoFont.WhiteSmallishText(), elementBounds2 = elementBounds2.BelowCopy())
 				.AddTextInput(elementBounds2.FlatCopy().WithAlignment(EnumDialogArea.RightFixed).WithFixedOffset(0.0, -3.0)
